coredns/resolver/resolver_suite_test.go

```diff
@@ -302,7 +303,7 @@ func newEndpointSlice(namespace, name, clusterID string, ports []mcsv1a1.Service
 
 	return &discovery.EndpointSlice{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      name + "-" + namespace + "-" + clusterID,
+			Name:      fmt.Sprintf("%s-%s-%s-%s", name, utilrand.String(5), namespace, clusterID),
```

Comment:
Need to release note this. Longer name may cause name length related failures for some deployments.

Reply:
yeah I thought about that. I think we should use the `GenerateName` functionality  that K8s uses for EndpointSlices with the service name as the prefix. I'm planning to do that next in a followup PR.